### PR TITLE
add option to define custom formatter to serialize version

### PR DESCRIPTION
**Pull request by marnikow**
If you want to serialize a version you don't have advanced options. By allowing a formatter the callee can reason about the version object and return a serialized version that fits his preference the best.

### CODE REVIEW

File: dunamai/formatters.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+from typing import Sequence, Union
+from dunamai import Version
+
+__all__ = ["pep440", "pep440_meta", "pep440_meta_id", "format_pieces_as_pep440"]
+
+
+def format_pieces_as_pep440(
```
Comment (mtkennerly) [inline dunamai/formatters.py:7]: You could just use the existing `serialize_pep440`. The only difference is that `format_pieces_as_pep440` doesn't call `check_version`, but that shouldn't hurt anything here.

File: dunamai/formatters.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        version.base,
+        stage=version.stage,
+        revision=version.revision,
+        post=version.distance,
+        epoch=version.epoch,
+        metadata=metadata,
+    )
+
+
+def pep440_meta(version: Version) -> str:
```
Comment (mtkennerly) [inline dunamai/formatters.py:84]: I don't think we need to have these formatters built-in, so please delete formatters.py. Dunamai handles these most common cases via boolean options in `serialize()`, which is more flexible than having separate functions for just a few specific combinations.

**Review by mtkennerly — CHANGES_REQUESTED** [review]

File: dunamai/__init__.py [hunk 3]
```diff
@@ -510,6 +512,23 @@ def serialize(
         check_version(out, style)
         return out
 
+    def serialize_with_formatter(
```
[the remark this answers is not in the document]
Comment (marnikow) [reply dunamai/__init__.py]: I updated the `serialize` function so that it accepts a callable as well.

**Review by mtkennerly — APPROVED** [review]
Thanks! There's a minor complaint from Mypy, but I can fix that and add some tests after the merge.

To avoid Mypy/etc issues in your other PRs, you can enable the pre-commit hooks by running this: `poetry run pre-commit install`

**Pull request merged**